pkg/ottl/contexts/internal/ottlcommon/span.go

```diff
@@ -18,6 +18,7 @@ import (
 	"context"
 	"encoding/hex"
 	"fmt"
+	"reflect"
```

Comment:
Is there way that this can be done without using the package reflect?

Reflect is a rather expensive package (in terms of increased static binary size and additional processing efforts)

Reply:
I will check about this.

Reply:
I can't speak to the efficiency of it, but an approach I prototyped while investigating arrays in the OTTL involved doing a type switch on the first element, and using generics to pass that type into another function that did the type conversion assertions on the individual array elements. I'm not familiar enough with what Go is doing under the hood, but it avoided use of the `reflect` package and I think should be fairly efficient.